docs/design/phases.md

```diff
@@ -0,0 +1,354 @@
+ - the `kube-dns` ServiceAccount is bound to the privileges in the `system:kube-dns` ClusterRole
+
+the TLS Bootstrap role (`system:node-bootstrapper`), so tokens in the group `kubeadm:kubelet-bootstrap` only can access what's required for TLS Bootstrapping
+
+## `kubeadm phase` UX
+
+ - `kubeadm phase certs selfsign`
+   - Inputs:
+     - `--dns-domain=cluster.local`
+     - `--service-cidr=10.96.0.1/12`
+     - `--alt-names=""`
+     - `--cert-dir=/etc/kubernetes/pki`
+   - Outputs (in the `--cert-dir` directory):
+     - `ca.{crt,key}`
+     - `apiserver.{crt,key}`
+     - `apiserver-kubelet-client.{crt,key}`
+     - `sa.{pub,key}`
+     - `front-proxy-ca.{crt,key}`
+     - `front-proxy-client.{crt,key}`
+ - `kubeadm phase kubeconfig clientcert`
+   - Inputs:
+     - `--client-name=`
+     - `--api-servers=`
+     - `--cert-dir=/etc/kubernetes/pki`
+     - `--organization=""`
+   - Output:
+     - The kubeconfig is `stdout`
+ - `kubeadm phase kubeconfig token`
+   - Inputs:
+     - `--client-name=`
+     - `--api-servers=`
+     - `--token`
+     - `--cert-dir=/etc/kubernetes/pki`
+   - Output:
+     - The kubeconfig is `stdout`
+ - `kubeadm phase controlplane etcd`
+   - Inputs:
+     - `--image=gcr.io/google_containers/etcd-${GOARCH}:3.0.14-kubeadm`
+     - `--data-dir=/var/lib/etcd`
+   - Output:
+     - `etcd.yaml` in the manifests dir (`/etc/kubernetes/manifests`)
+ - `kubeadm phase controlplane apiserver`
+   - Inputs:
+     - `--image=gcr.io/google_containers/kube-apiserver-${ARCH}:${STABLE_VERSION}`
+     - `--service-cidr=10.96.0.1/12`
+     - `--cert-dir=/etc/kubernetes/pki`
+     - `--port=6443`
+     - `--address=0.0.0.0`
+     - `--etcd-servers=http://localhost:2379`
+     - `--authorization-mode=RBAC`
+     - `--extra-args`
+     - `--extra-volumes`
+   - Output:
+     - `kube-apiserver.yaml` in the manifests dir (`/etc/kubernetes/manifests`)
+ - `kubeadm phase controlplane controller-manager`
+   - Inputs:
+     - `--image=gcr.io/google_containers/kube-controller-manager-${ARCH}:${STABLE_VERSION}`
+     - `--kubeconfig=/etc/kubernetes/controller-manager.conf`
+     - `--pod-network-cidr`
+     - `--cert-dir=/etc/kubernetes/pki`
+     - `--auto-approve-csrs=true`
+     - `--extra-args`
+     - `--extra-volumes`
+   - Output:
+     - `kube-controller-manager.yaml` in the manifests dir (`/etc/kubernetes/manifests`)
+ - `kubeadm phase controlplane scheduler`
+   - Inputs:
+     - `--image=gcr.io/google_containers/kube-scheduler-${ARCH}:${STABLE_VERSION}`
+     - `--kubeconfig=/etc/kubernetes/scheduler.conf`
+     - `--extra-args`
+   - Output:
+     - `kube-scheduler.yaml` in the manifests dir (`/etc/kubernetes/manifests`)
+ - `kubeadm phase addons cluster-info`
+   - Inputs:
+     - `--admin-kubeconfig=/etc/kubernetes/admin.conf`
+   - Outputs:
+     - Creates the `cluster-info` ConfigMap in the `kube-public` namespace
+     - Exposes it publicly with RBAC bindings
+- `kubeadm phase addons self-hosted`
```

Comment:
might be that considering `self-hosted` as an `addons` can be confusing for users.
I suggest to treat `self-hosted` phase separately from `addons`